core/extract_data/core/nixpkgs-graph.nix

```diff
@@ -0,0 +1,125 @@
+          output_paths =
+            let
+              outPaths =
+                let
+                  outputs =
+                    let
+                      pEvalResult = builtins.tryEval (toString (if okValue ? outputs then okValue.outputs else ""));
+                    in
+                    if pEvalResult.success then splitString " " pEvalResult.value
+                    else [ ];
+                in
+                map
+                  (p:
+                    {
+                      name = p;
+                      value = (builtins.tryEval (if okValue ? ${p} then toString okValue.${p} else "")).value;
+                    }
+                  )
+                  (outputs);
+            in
+            builtins.listToAttrs outPaths;
+
+          nixpkgs_metadata =
+            {
+              pname = (builtins.tryEval (if okValue ? name && okValue.name != false
+              then (builtins.parseDrvName okValue.name).name
+              else "")).value;
+              version = (builtins.tryEval (if okValue ? name && okValue.name != false
+              then (builtins.parseDrvName okValue.name).version
+              else "")).value;
+              broken = (builtins.tryEval (if okValue ? meta.broken
+              then okValue.meta.broken
+              else false)).value;
+              license = (builtins.tryEval (if okValue ? meta.license.fullName
+              then okValue.meta.license.fullName
+              else "")).value;
+            };
+
+          build_inputs = builtins.concatLists
+            [
+              (remove null
+                (map
+                  (p:
+                    let
+                      pEvalResult = builtins.tryEval (p);
+                    in
+                    if pEvalResult.success && (toString pEvalResult.value) != "" then
+                      {
+                        build_input_type = "build_input";
+                        package = recurse "" "" pEvalResult.value;
+                      }
+                    else null
+                  )
+                  (okValue.buildInputs or [ ])))
+
+              (remove null
+                (map
+                  (p:
+                    let
+                      pEvalResult = builtins.tryEval (p);
+                    in
+                    if pEvalResult.success && (toString pEvalResult.value) != "" then
+                      {
+                        build_input_type = "propagated_build_input";
+                        package = recurse "" "" pEvalResult.value;
+                      }
+                    else null
+                  )
+                  (okValue.propagatedBuildInputs or [ ])))
+            ];
+
+        }
+      else
+
+        if isAttrs okValue && (okValue.recurseForDerivations or false) then
+          mapAttrs (recurse path) okValue
+        else null
+    else null;
+
+in
+
+{
+  packages =
+    (collect
```

Comment:
If we use "recurse" there

https://github.com/tweag/nixpkgs-graph-explorer/pull/76/files#diff-9d7a87712ef06fa0874a93d54288db29db414e699ef7fc3bfd2bb58a72d66bb3R87

then we don't want to collect here

Reply:
@GuillaumeDesforges Is this collect meant to collect all the packages info in targetFlakePkgs, as long as it has output_paths attribute?

Reply:
Ah yes you are right, I don't need to make the recursive function "recurse" again